flow/actorcompiler/ActorParser.cs

```diff
@@ -255,6 +255,71 @@ public ActorParser(string text, string sourceFile, ErrorMessagePolicy errorMessa
             //showTokens();
         }
 
+        class ClassContext {
+            public string name;
+            public int inBlocks;
+        }
+
+        private bool ParseClassContext(TokenRange toks, out string name)
+        {
+            name = "";
+            if (toks.Begin == toks.End)
+            {
+                return false;
+            }
+
+            // http://nongnu.org/hcb/#attribute-specifier-seq
+            Token first;
+            while (true)
+            {
+                first = toks.First(NonWhitespace);
+                if (first.Value == "[")
+                {
+                    var contents = first.GetMatchingRangeIn(toks);
+                    toks = range(contents.End + 1, toks.End);
+                }
+                else if (first.Value == "alignas")
+                {
+                    toks = range(first.Position + 1, toks.End);
+                    first = toks.First(NonWhitespace);
+                    first.Assert("Expected ( after alignas", t => t.Value == "(");
+                    var contents = first.GetMatchingRangeIn(toks);
+                    toks = range(contents.End + 1, toks.End);
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            // http://nongnu.org/hcb/#class-head-name
+            first = toks.First(NonWhitespace);
+            if (!identifierPattern.Match(first.Value).Success) {
+                return false;
+            }
+            while (true) {
+                first.Assert("Expected identifier", t=>identifierPattern.Match(t.Value).Success);
+                name += first.Value;
+                toks = range(first.Position + 1, toks.End);
+                if (toks.First(NonWhitespace).Value == "::") {
+                    name += "::";
+                    toks = toks.SkipWhile(Whitespace).Skip(1);
+                } else {
+                    break;
+                }
+                first = toks.First(NonWhitespace);
+            }
+            // http://nongnu.org/hcb/#class-virt-specifier-seq
+            toks = toks.SkipWhile(t => Whitespace(t) || t.Value == "final" || t.Value == "explicit" || t.Value == "sealed");
```

Comment:
Sealed is non-standard, but used in our codebase.  Bleh, but good catch.  We should probably do a change to s/sealed/final/ and remove sealed from Platform.h.